Nimda/Core/PluginContainer.php

```diff
@@ -129,16 +129,41 @@ public function onMessage(Message $message)
             return;
         }
 
-        $plainText = trim(Str::after($plainText, $plugin->keys()->first()));
+        $arguments = $this->parseArguments($plainText, $plugin->keys()->first());
+
+        if($arguments === false) {
+            return;
+        }
 
         $plugin = $plugin->first();
-        $plugin->trigger($message, $plainText);
+        $plugin->trigger($message, $arguments);
     }
 
+    /**
+     * @param $text
+     * @return Collection
+     */
     private function findPluginByCommand($text)
     {
         return $this->commands->filter(function ($plugin) use ($text){
-            return Str::startsWith($text, array_keys($plugin)[0]);
+            $commandShard = explode(' ', array_keys($plugin)[0])[0];
```

Comment:
Is there any better way to this line?

Reply:
Not entirely sure. This method of doing it, is really bad though.